graylog2-server/src/main/java/org/graylog2/indexer/Deflector.java

```diff
@@ -182,29 +180,31 @@ public void cycle() {
     }
 
     public int getNewestTargetNumber() throws NoTargetIndexException {
-        final Map<String, IndexStats> indices = this.indices.getAll();
-        if (indices.isEmpty()) {
+        final Set<String> indexNames = indices.getIndexNamesAndAliases(getDeflectorWildcard()).keySet();
+
+        if (indexNames.isEmpty()) {
             throw new NoTargetIndexException();
         }
 
-        final List<Integer> indexNumbers = Lists.newArrayListWithExpectedSize(indices.size());
-        for (String indexName : indices.keySet()) {
+        int highestIndexNumber = -1;
+        for (String indexName : indexNames) {
             if (!isGraylogDeflectorIndex(indexName)) {
                 continue;
             }
 
             try {
-                indexNumbers.add(extractIndexNumber(indexName));
+                final int indexNumber = extractIndexNumber(indexName);
+                highestIndexNumber = Math.max(indexNumber, highestIndexNumber);
             } catch (NumberFormatException ex) {
-                LOG.debug("Couldn't extract index number from index name " + indexName, ex);
+                LOG.warn("Couldn't extract index number from index name " + indexName, ex);
             }
         }
 
```

Comment:
What about replacing this with a streams call like this:

```
        final Optional<Integer> highestIndexNumber = indexNames.stream()
            .filter(indexName -> !this.isGraylogDeflectorIndex(indexName))
            .map(Deflector::extractIndexNumber)
            .max(Integer::max);
```
Makes it easier to grasp what is done (at least for me).